frontend/common/APICall.js

```diff
@@ -0,0 +1,77 @@
+import { Links } from "./constants.js";
+class APICall{
+
+    constructor(){
+        this.authToken = localStorage.getItem('authToken');
+        this.headers = new Headers({
+            'Authorization': `Bearer ${this.authToken}`,
+            'Content-Type': 'application/json',
+        });
+    }
+    
+    makeGetRequest(path) {
+
+        const options = {
+            method: 'GET',
+            headers: this.headers,
+        };
+
+        return new Promise((resolve, reject) => {
+            fetch(`${Links.serverBaseURL}/${path}`, options)
+            .then(response => {
+                if (!response.ok) {
+                    if(response.status === 401){
+                        localStorage.setItem('signedIn', "false");
+                        window.location.replace('index.html');
+                    } else if(response.status === 404){
+                        window.location.replace('404.html');
+                    } else {
+                        reject(new Error('Network response was not ok'));
+                    }
+                    return;
+                }
+                return response.json();
+            })
+            .then(data => {
+                resolve(data);
+            })
+            .catch(error => {
+                reject(error);
+            });
+        });
+    }
+
+    makePostRequest(path, body) {
```

Comment:
similar to the previous function, let's name it to something more descriptive

Reply:
name makes sense